### PR TITLE
Option to send transcripts when reporting user

### PR DESCRIPTION
**Pull request by zapu**

### CODE REVIEW

File: go/service/user_report.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+func pullTranscript(mctx libkb.MetaContext, chatG *globals.ChatContext, convIDStr string) (res string, err error) {
+	convIDBytes, err := chat1.MakeConvID(convIDStr)
+	if err != nil {
+		return "", err
+	}
+	uidBytes := gregor1.UID(mctx.CurrentUID().ToBytes())
+	chatQuery := &chat1.GetThreadQuery{
+		MarkAsRead: false,
+		MessageTypes: []chat1.MessageType{
```
Comment (mmaxim) [inline go/service/user_report.go]: Just pass in `VisibleMessageTypes`

File: go/service/user_report.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		}
+		mv := msg.Valid()
+		mb := mv.MessageBody
+		var textMsg *convTranscriptMsgText
+		if mb.IsType(chat1.MessageType_TEXT) {
+			textMsg = &convTranscriptMsgText{
+				Body: mb.Text().Body,
+			}
+		}
+		tMsg := convTranscriptMsg{
```
Comment (mmaxim) [inline go/service/user_report.go]: Can't you use `json.Marshal` to make this easier?

Comment (zapu) [reply go/service/user_report.go]: Should we marshal the entire threadView? I wanted to only pass essential fields to save bandwidth and space.

Comment (mmaxim) [reply go/service/user_report.go]: Just the `MessageBody` portion of the messages.

File: go/service/user_report.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	}
+	uidBytes := gregor1.UID(mctx.CurrentUID().ToBytes())
+	chatQuery := &chat1.GetThreadQuery{
+		MarkAsRead:   false,
+		MessageTypes: chat1.VisibleChatMessageTypes(),
+	}
+	pagination := &chat1.Pagination{
+		Num: transcriptMessageLimit,
+	}
+	threadView, err := chatG.ConvSource.Pull(mctx.Ctx(), convIDBytes, uidBytes, chat1.GetThreadReason_GENERAL,
```
Comment (zapu) [inline go/service/user_report.go]: @mmaxim should we have a new `GetThreadReason` for this?

Comment (mmaxim) [reply go/service/user_report.go]: Nah, this is going to be a vanishingly small number of the overal reqs

File: go/service/user_report.go (hunk 2 above)
Comment (zapu) [reply go/service/user_report.go]: both comments addressed - good for another look

**Pull request merged**